cmd/version.go

```diff
@@ -0,0 +1,102 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/kris-nova/kubicorn/cutil/logger"
+	"github.com/spf13/cobra"
+)
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Verify Kubicorn version",
+	Long: `Use this command to check the version of Kubicorn.
+
+This command will return the version of the Kubicorn binary.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := RunVersion(vo)
+		if err != nil {
+			logger.Critical(err.Error())
+			os.Exit(1)
+		}
+
+	},
+}
+
+// VersionOptions contains fields for version output
+type VersionOptions struct {
+	Version   string `json:"Version"`
+	GitCommit string `json:"GitCommit"`
+	BuildDate string `json:"BuildDate"`
+	GOVersion string `json:"GOVersion"`
+	GOARCH    string `json:"GOARCH"`
+	GOOS      string `json:"GOOS"`
+}
+
+var vo = &VersionOptions{}
+
+func init() {
+	RootCmd.AddCommand(versionCmd)
+}
+
+// RunVersion populates VersionOptions and prints to stdout
+func RunVersion(vo *VersionOptions) error {
+
+	vo.Version = getVersion()
+	vo.GitCommit = getGitCommit()
+	vo.BuildDate = time.Now().UTC().String()
+	vo.GOVersion = runtime.Version()
+	vo.GOARCH = runtime.GOARCH
+	vo.GOOS = runtime.GOOS
+	voBytes, err := json.Marshal(vo)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Kubicorn version: ", string(voBytes))
+	return nil
+}
+
+var (
+	versionFile = "/src/github.com/kris-nova/kubicorn/VERSION"
```

Comment:
As I said, I'm trying to make it work for the first time in every case. :D Maybe we should keep it simple. 

Yes, you're right for `go get`, you can't use `ldflags`. But with `go get` your code will work perfectly.
For `go build` and `make`, yes it uses `ldflags`, but if somebody uses `kubicorn version`, it can get wrong results? Why, because if it forgot to run build, it would show him latest `git` hash, while he's not on that version.
And as I mentioned, as far as I understood, we will release executables, that somebody would be able to download from GitHub release pages. In that case, without `go` installed, `kubicorn version` would fail due to Go/`git` stuff not being available.

Hope you understand better. :D